test/e2e/operator_test.go

```diff
@@ -1076,6 +1077,303 @@ func TestContainerLogging(t *testing.T) {
+			Policy: operatorv1.AppendHTTPHeaderPolicy,
+		},
+	}
+	if err := kclient.Update(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+	if err := waitForDeploymentEnvVar(kclient, deployment, 1*time.Minute, "ROUTER_SET_FORWARDED_HEADERS", "append"); err != nil {
+		t.Fatalf("failed to observe ROUTER_SET_FORWARDED_HEADERS=append: %v", err)
+	}
+	if err := waitForDeploymentComplete(kclient, deployment, 3*time.Minute); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+	testRoute(1, nil)
+	testRoute(3, []string{"x-forwarded-for:foo", "x-forwarded-for:bar"})
+
+	// Verify that we get the expected behavior if we set the policy to
+	// "replace".  We should always get 1 X-Forwarded-For header.
+	if err := kclient.Get(context.TODO(), icName, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller: %v", err)
+	}
+	ic.Spec.HTTPHeaders = &operatorv1.IngressControllerHTTPHeaders{
+		Forwarded: &operatorv1.IngressControllerHTTPForwardedHeaderPolicy{
+			Policy: operatorv1.ReplaceHTTPHeaderPolicy,
+		},
+	}
+	if err := kclient.Update(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+	if err := waitForDeploymentEnvVar(kclient, deployment, 1*time.Minute, "ROUTER_SET_FORWARDED_HEADERS", "replace"); err != nil {
+		t.Fatalf("failed to observe ROUTER_SET_FORWARDED_HEADERS=replace: %v", err)
+	}
+	if err := waitForDeploymentComplete(kclient, deployment, 3*time.Minute); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+	testRoute(1, nil)
+	testRoute(1, []string{"x-forwarded-for:foo", "x-forwarded-for:bar"})
+
+	// Verify that we get the expected behavior if we set the policy to
+	// "never".  We should always receive exactly as many X-Forwarded-For
+	// headers as we send.
+	if err := kclient.Get(context.TODO(), icName, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller: %v", err)
+	}
+	ic.Spec.HTTPHeaders = &operatorv1.IngressControllerHTTPHeaders{
+		Forwarded: &operatorv1.IngressControllerHTTPForwardedHeaderPolicy{
+			Policy: operatorv1.NeverHTTPHeaderPolicy,
+		},
+	}
+	if err := kclient.Update(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+	if err := waitForDeploymentEnvVar(kclient, deployment, 1*time.Minute, "ROUTER_SET_FORWARDED_HEADERS", "never"); err != nil {
+		t.Fatalf("failed to observe ROUTER_SET_FORWARDED_HEADERS=never: %v", err)
+	}
+	if err := waitForDeploymentComplete(kclient, deployment, 3*time.Minute); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+	testRoute(0, nil)
+	testRoute(2, []string{"x-forwarded-for:foo", "x-forwarded-for:bar"})
+
+	// Verify that we get the expected behavior if we set the policy to
+	// "if-none".  We should always receive at least 1 X-Forwarded-For
+	// header, and if the request specifies more than 1, we should receive
+	// the number that are in the request.
```

Comment:
```suggestion
	// the number of headers specified in the request.
```

Does that make more sense, or am I missing something?

Reply:
I've updated the wording to be more explicit.